### ISSUE TITLE
Occasionally encounters `Framerate Exception: Exiting Program.`

### ISSUE THREAD

**Issue by gordykeene** [EnigmaDragons/MonoGameJam#348]
As best I can tell, when `OnOverwatchSelected` iterates over the list of tiles, something in `ShotCalculation(...).BestShot()` causes the framerate to drop.

In my Output Debug window, I saw this:
```
Framerate Warning: Framerate = 10
Framerate Warning: Framerate = 6
Framerate Warning: Framerate = 4
Framerate Warning: Framerate = 8
Framerate Warning: Framerate = 10
Framerate Warning: Framerate = 10
Framerate Warning: Framerate = 10
Framerate Warning: Framerate = 10
Framerate Warning: Framerate = 4
Framerate Warning: Framerate = 2
Framerate Warning: Framerate = 2
Framerate Warning: Framerate = 2
Framerate Warning: Framerate = 4
Framerate Warning: Framerate = 2
Framerate Warning: Framerate = 2
Framerate Warning: Framerate = 2
Framerate Warning: Framerate = 4
Framerate Warning: Framerate = 4
Framerate Exception: Exiting Program.
```
In game, my machine usually shows 60 FPS.

**Comment by gordykeene** [comment]
I added some trivial timer code, and found it takes 52+ seconds to loop over the tiles.

```cs
                var sw = new System.Diagnostics.Stopwatch();
                sw.Start();
                tiles.ForEach(x =>
                {
                    var shot = new ShotCalculation(GameWorld.CurrentCharacter.CurrentTile, GameWorld.Map[x]).BestShot();
                    if (new HitChanceCalculation(GameWorld.CurrentCharacter.Accuracy, shot.BlockChance).Get() > 0)
                        overwatchedTiles[x] = shot;
                });
                sw.Stop();
                System.Console.WriteLine($"Elapsed={sw.Elapsed}");
```

**Comment by SilasReinagel** [comment]
Probably solved by EnigmaDragons/MonoGameJam#343

**Comment by timeracers** [comment]
Do you experience any Framerate warnings any more?
